test/unit/resolve.js

```diff
@@ -1,175 +1,185 @@
 /* global describe it beforeEach */
 
+var _ = require('lodash')
 var proxyquire = require('proxyquire')
 var assert = require('assert')
 var mock = require('mockmock')
 var EventEmitter = require('events').EventEmitter
+var config = require('../../lib/config')
 
 describe('resolve', function () {
-  var resolve
-  var http
-  var cb
-  var httpResp
-  var httpGet
-
-  beforeEach(function () {
-    httpResp = new EventEmitter()
-    httpGet = new EventEmitter()
-    http = { get: mock(httpGet) }
-    cb = mock()
-
-    resolve = proxyquire('../../lib/resolve', {
-      http: http
+  var protocols = ['http', 'https']
+  protocols.forEach(function (protocol) {
+    describe(protocol, function () {
+      var http
+      var cb
+      var httpResp
+      var httpGet
+      var resolve
+      beforeEach(function () {
+        httpResp = new EventEmitter()
+        httpGet = new EventEmitter()
+        http = { get: mock(httpGet) }
+        var dummyConfig = _.clone(config)
```

Comment:
I want to avoid adding lodash, but would rather have smaller, isolated dependencies. `object-assign` is already in there, so we should be able to do this:

```js
var assign = require('object-assign')
var dummyConfig = assign({}, config)
```